bundles/binding/org.openhab.binding.sallegra/src/main/java/org/openhab/binding/sallegra/internal/SallegraNode.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sallegra.internal;
+
+import org.openhab.binding.sallegra.internal.xml.XmlUtils;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Command;
+import org.openhab.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Node structure to kindly represent a Sallegra Module
+ * 
+ * @author Benjamin Marty (Developed on behalf of Satelco.ch)
+ * @since 1.8.0
+ * 
+ */
+public class SallegraNode {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(SallegraNode.class);
+
+	private static final String GET = "GET";
+
+	/*
+	 * Suffix for specific recurring Parts in Sallegra Modules
+	 */	
+	private static final String SUFFIX_PASSWORD = "pw=";
+	private static final String XML_PATH = "current_state.xml";
+
+	/*
+	 * Suffix for specific recurring Parts in URL gen
+	 */	
+	private static final String SUFFIX_DIMMER = "Dimmer";
+	private static final String SUFFIX_RELAY = "Relay";
+	
+	
+	private String hostName;
+	private String password;
+	private int timeOut = 10000;
+
+	/*
+	 * Getter
+	 */
+	public String getHostName() {
+		return hostName;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+	
+	
+	protected void setHostName(String hostName) {
+		this.hostName = hostName;
+	}
+
+	protected void setPassword(String password) {
+		this.password = password;
+	}
+	
+	/**
+	 * Properly configured if hostname & password is set
+	 */
+	public boolean properlyConfigured() {
+		return (hostName != null && password != null) || timeOut != 0;
+	}
+
+	
+	/**
+	 * Sets the State of the Relay on the Relay Module
+	 */
+	public void setRelay(Command command, String number) {
+		// Check if Instance is Digital OnOff Value
+		if (command instanceof OnOffType) {
+			if( ((OnOffType) command) == OnOffType.ON) {
+				HttpUtil.executeUrl(GET, createURL()
```

Comment:
Please extrace the generated url into a variable before calling HttpUtil.executeUrl(), this makes debugging much easier.